packages/blade/src/components/Tour/Tour.web.tsx

```diff
@@ -0,0 +1,199 @@
+  const [size, setSize] = useState<TourMaskRect>({
+    x: 0,
+    y: 0,
+    height: 0,
+    width: 0,
+  });
+
+  // delayed state is used to let the transition finish before reacting to the state changes
+  const delayedActiveStep = useDelayedState(activeStep, transitionDelay);
+  const delayedSize = useDelayedState(size, transitionDelay);
+  // keep track of when we are transitioning between steps
+  const isTransitioning = useIsTransitioningBetweenSteps(activeStep, transitionDelay);
+
+  const currentStepRef = refIdMap.get(steps[delayedActiveStep]?.name);
+  const intersection = useIntersectionObserver(currentStepRef!, {
+    threshold: 0.5,
+  });
+
+  // main step logic
+  const totalSteps = steps.length;
+  const currentStepData = useMemo(() => {
+    return steps[activeStep];
+  }, [activeStep, steps]);
+
+  const goToNext = useCallback(() => {
+    let nextState = activeStep + 1;
+    if (nextState >= steps.length) {
+      nextState = steps.length - 1;
+    }
+    onStepChange?.(nextState);
+  }, [activeStep, onStepChange, steps.length]);
+
+  const goToPrevious = useCallback(() => {
+    let nextState = activeStep - 1;
+    if (nextState < 0) {
+      nextState = 0;
+    }
+    onStepChange?.(nextState);
+  }, [activeStep, onStepChange]);
+
+  const stopTour = useCallback(() => {
+    onFinish?.();
+  }, [onFinish]);
+
+  const attachStep = useCallback((id: string, ref: React.RefObject<HTMLElement>) => {
+    if (!ref) return;
+    setRefIdMap((prev) => {
+      return new Map(prev).set(id, ref);
+    });
+  }, []);
+
+  const removeStep = useCallback((id: string) => {
+    setRefIdMap((prev) => {
+      const newMap = new Map(prev);
+      newMap.delete(id);
+      return newMap;
+    });
+  }, []);
+
+  const updateMaskSize = useCallback(() => {
+    const ref = refIdMap.get(steps[activeStep]?.name);
+    if (!ref?.current) return;
+
+    const rect = ref.current.getBoundingClientRect();
+    setSize({
+      x: rect.x,
+      y: rect.y,
+      width: rect.width,
+      height: rect.height,
+    });
+  }, [activeStep, refIdMap, steps]);
+
+  const scrollToStep = useCallback(() => {
+    const ref = refIdMap.get(steps[delayedActiveStep]?.name);
+    if (!ref?.current) return;
+
+    // If the element is already in view, don't scroll
+    if (intersection?.isIntersecting) return;
+
+    smoothScroll(ref.current, {
+      behavior: 'smooth',
+      block: 'center',
+      inline: 'center',
+    })
+      .then(() => {
+        updateMaskSize();
+      })
+      .finally(() => {
+        // do nothing
+      });
+  }, [delayedActiveStep, refIdMap, steps, updateMaskSize, intersection?.isIntersecting]);
+
+  // Update the size of the mask when the active step changes
+  useIsomorphicLayoutEffect(() => {
+    updateMaskSize();
+  }, [activeStep, refIdMap, steps, updateMaskSize]);
+
+  // scroll into view when the active step changes
+  // What a time to hit a chrome bug: https://bugs.chromium.org/p/chromium/issues/detail?id=1043933
```

Comment:
Is this affecting us? Do we have any workaround? How bad does it look?

Reply:
> How bad does it look?

Very bad. 

But fixed it by other means. Check on [slack](https://razorpay.slack.com/archives/G01B3LQ9H0W/p1699515674579959).